### PR TITLE
Elaborate on shallow submodule options

### PR DESCRIPTION
**Pull request by zyphlar**
I often clone things first and read docs second rather than use Github's interface. This causes problems with submodules, especially shallow submodules. I finally found a workflow that's pretty ok and figured I'd share it here for other contributors.

### CODE REVIEW

File: docs/INSTALL.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ If you're only doing a one-off build or your internet bandwidth or disk space is
 
 - a `--shallow-submodules` option to _shallow clone_ the submodules (save ~1.3Gb) - this is suitable for a generic development if no work on submodules is planned.
 
+If you mistakenly did a `git clone` without checking out submodules, you can run `git submodule update --init --recursive`. If you want to do so shallowly, you can run `git submodule foreach 'git config -f .gitmodules submodule.$sm_path.shallow true'` beforehand and/or add `--depth=1` to the update command.
```
Comment (biodranik) [inline docs/INSTALL.md:75]: What is the benefit of the longer foreach vs a shorter --depth=1 ? If there are no benefits, let's leave only a shorter option.

**Comment by zyphlar** [comment]
I would need a git expert to comment, I think sometimes things can get half checked out or half initialized but I do notice that it seems the shallow setting may have not been necessary

May 26, 2024 2:20:07 PM Alexander Borsuk ***@***.***>:

File: docs/INSTALL.md (hunk 1 above)
Comment (zyphlar) [reply docs/INSTALL.md:75]: @biodranik response below

Comment (biodranik) [reply docs/INSTALL.md:75]: I've googled and found that .shallow setting does a different thing https://stackoverflow.com/a/37933909

Comment (zyphlar) [reply docs/INSTALL.md:75]: OK maybe it's not necessary for this use case, I'll remove.

File: docs/INSTALL.md [hunk 2]
```diff
@@ -72,6 +72,8 @@ If you're only doing a one-off build or your internet bandwidth or disk space is
 
 - a `--shallow-submodules` option to _shallow clone_ the submodules (save ~1.3Gb) - this is suitable for a generic development if no work on submodules is planned.
 
+If you mistakenly did a `git clone` without checking out submodules, you can run `git submodule update --init --recursive`. If you want to do so shallowly, you can add `--depth=1` to the update command.
```
Comment (biodranik) [inline docs/INSTALL.md:75]:
```suggestion
If you mistakenly did a `git clone` without checking out submodules, you can run `git submodule update --init --recursive`. If you don't want to clone complete submodules, you can add `--depth=1` to the update command.
```

**Pull request merged**